plugins/search-indexer-backend/src/types.ts

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Base properties that all indexed documents must include, as well as some
+ * common properties that documents are encouraged to use where appropriate.
+ */
+export interface IndexableDocument {
+  /**
+   * The primary name of the document (e.g. name, title, identifier, etc).
+   */
+  title: string;
+
+  /**
+   * Free-form text of the document (e.g. description, content, etc).
+   */
+  text: string;
+
+  /**
+   * The relative or absolute URL of the document (target when a search result
+   * is clicked).
+   */
+  location: string;
+
+  /**
+   * (Optional) The owner of the document (e.g. spec.owner on a catalog entity).
+   */
+  owner?: string;
+
+  /**
+   * (Optional) The lifecycle of the document (e.g. spec.lifecycle on a catalog entity).
+   */
+  lifecycle?: string;
```

Comment:
Bit scared about getting this detailed with the base type, wondering if these kinda things shouldn't be added through decorators or smth?

Reply:
For sure.  We want to keep decoupled from the catalog, but still provide some strong suggestions on what most things in search may have (so that cross-index filtering is more meaningful).

I see decorators as ways that Backstage instance owners can use to apply very generic things that are specific to their org (say, a high-level quality score, or a high-level usage score).

So maybe `lifecycle` isn't the right thing, here.  But is there something else catalog-ish that can easily be imagined as a cross-index filter?